addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2681,6 +2681,21 @@ options.registry.ScrollButton = options.Class.extend({
     // Options
     //--------------------------------------------------------------------------
 
+     /**
+     * @override
+     */
+     async selectClass(previewMode, widgetValue, params) {
+         await this._super(...arguments);
+         if (params.isFullHeight) {
+             this.$button.show();
+         } else {
+             if (previewMode) {
+                 this.$button.hide();
+             } else {
+                 this.$button.detach();
+             }
+         }
```

Comment:
This should only apply to the specific height class control, not all potential selec-class that could be added in that option. You can handle that by using two methods on the same we-button (so here data-select-class for the height + data-... to handle the button hide/detach).